geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/ConvertToGeoFile.scala

```diff
@@ -0,0 +1,238 @@
+/***********************************************************************
+ * Copyright (c) 2015-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.geomesa.nifi.datastore.processor
+
+import java.io.{File, OutputStream}
+import java.nio.file.Files
+import java.util.UUID
+import java.util.zip.GZIPOutputStream
+
+import org.apache.commons.io.FilenameUtils
+import org.apache.nifi.annotation.behavior.{WritesAttribute, WritesAttributes}
+import org.apache.nifi.annotation.documentation.{CapabilityDescription, Tags}
+import org.apache.nifi.components.PropertyDescriptor
+import org.apache.nifi.expression.ExpressionLanguageScope
+import org.apache.nifi.flowfile.FlowFile
+import org.apache.nifi.processor._
+import org.apache.nifi.processor.io.OutputStreamCallback
+import org.apache.nifi.processor.util.StandardValidators
+import org.geomesa.nifi.datastore.processor.mixins.ConvertInputProcessor
+import org.geomesa.nifi.datastore.processor.mixins.ConvertInputProcessor.ConverterCallback
+import org.geomesa.nifi.datastore.processor.validators.GzipLevelValidator
+import org.geotools.util.factory.Hints
+import org.locationtech.geomesa.index.conf.QueryHints
+import org.locationtech.geomesa.tools.`export`.formats.FeatureExporter.ExportStream
+import org.locationtech.geomesa.tools.`export`.formats.FileSystemExporter.{OrcFileSystemExporter, ParquetFileSystemExporter}
+import org.locationtech.geomesa.tools.`export`.formats._
+import org.locationtech.geomesa.utils.io.{CloseWithLogging, PathUtils}
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.util.control.NonFatal
+
+@Tags(Array("OGC", "geo", "convert", "converter", "simple feature", "geotools", "geomesa",
+  "parquet", "orc", "arrow", "gml", "geojson", "csv", "leaflet"))
+@CapabilityDescription("Convert incoming files into OGC data files using GeoMesa Converters")
+@WritesAttributes(
+  Array(
+    new WritesAttribute(attribute = "geomesa.convert.successes", description = "Number of features written successfully"),
+    new WritesAttribute(attribute = "geomesa.convert.failures", description = "Number of features with errors")
+  )
+)
+class ConvertToGeoFile extends ConvertInputProcessor {
```

Comment:
There's part of me that wants to say that we ought to follow the GeoTools OGR DataStore approach and just offer up a conversion that's re-use what GDAL supports.

We'd get all these formats: https://gdal.org/drivers/vector/index.html  Surely that'd make everything happy!?!?!

Reply:
it supports excel so i'm on board!

Reply:
Le sigh, no PPT support.  Not *everyone* will be happy....  like usual.